From a323fd626627e8d42819fd5b7e2829196b5c54a3 Mon Sep 17 00:00:00 2001
From: Matteo Beccati <matteo@beccati.com>
Date: Fri, 20 Nov 2015 07:48:59 +0100
Subject: [PATCH] Fix h1 report 99004

Reflected XSS
-------------

The HackerOne user @decidedlygray has reported that the affiliate-preview.php
script in www/admin is vulnerable to a reflected XSS attack via the "charset"
parameter. If the target system is running a Revive Adserver version older
than 3.2.2, such a vulnerability could be used by an attacker to steal the
session ID of an authenticated user, by tricking them into visiting a
specifically crafted URL.

A CVE-ID has been requested, but not assigned yet.

CWE: CWE-79
CVSSv2: 3.5 (AV:N/AC:M/Au:S/C:N/I:P/A:N)
---
 .../plugins/invocationTags/oxInvocationTags/spc.class.php       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins_repo/openXInvocationTags/plugins/invocationTags/oxInvocationTags/spc.class.php b/plugins_repo/openXInvocationTags/plugins/invocationTags/oxInvocationTags/spc.class.php
index 5674ae3bd1..36a8cd5316 100644
--- a/plugins_repo/openXInvocationTags/plugins/invocationTags/oxInvocationTags/spc.class.php
+++ b/plugins_repo/openXInvocationTags/plugins/invocationTags/oxInvocationTags/spc.class.php
@@ -233,7 +233,7 @@ function selectElement() {
                         break;
                 case 'charset':
                         $optionName = $GLOBALS['strCharset'];
-                        $optionValue = empty($mi->$feature) ? $GLOBALS['strAutoDetect'] : $mi->$feature;
+                        $optionValue = empty($mi->$feature) ? $GLOBALS['strAutoDetect'] : htmlspecialchars($mi->$feature);
                         break;
                 default:
                         $optionName = $feature;
